proxy/http/HttpTransactCache.cc

```diff
@@ -1299,7 +1299,7 @@ HttpTransactCache::match_response_to_request_conditionals(HTTPHdr *request, HTTP
       ink_time_t lm_value = response->get_last_modified();
 
       // we won't return NOT_MODIFIED if Last-modified is too recent
-      if ((lm_value == 0) || (request->get_if_modified_since() < lm_value)) {
```

Comment:
This exact logic - forcing 304 on origin 200 wheb last-modified = if-modified-since - breaks Akamai live streaming.

Their logic is simple: if-modified-since <timestamp> means "give me updates on this stream since <timestamp>". response is 200 OK, last-modified <same_timestamp>, which means "here's what's new since <timestamp>".

Currently you can't watch akamized live streams via ATS in forwarding proxy mode, as from second request onwards it uses if-modified=last-modified request/response logic for stream updates.

This is why my suggestion is to treat the edge case of if-modified=last-modified as HTTP_STATUS_OK and allow client to get the new body.